src/core/cluster.py

```diff
@@ -59,7 +59,10 @@ def peer_relation(self) -> Relation | None:
     @property
     def opensearch_relation(self) -> Relation | None:
         """The Opensearch Server relation."""
-        return self.model.get_relation(OPENSEARCH_REL_NAME)
+        try:
+            return self.model.get_relation(OPENSEARCH_REL_NAME)
+        except KeyError:
```

Comment:
Yes, I encountered `KeyError`...

I'll give it a try to reproduce and then raise the bug. However until the bug is fixed, we should make sure that our code is stable, so we should leave the check.

Reply:
I tried, but so far I can't reproduce the issue. However I remember that I added this code for a reason. It would be difficult to report anything to `operator` w/o more precision.

Therefore we have 3 choices.

1. We remove the check and add it back once the error was observed again. (I don't know the risks for this solution with GE around...)
2. We keep the --otherwise harmless-- check, perhaps with logging added. This way we can track if the case occurs, and we're protected in case it may.
3. We open a separate ticket on investigating the issue (as it IS a unit of work, more than 30 mins)

I'm sorry to have let this one go (I didn't realize at the moment that it was a bug).

However, in my defense I'm spending non-negligable time lately on creating (descriptive) bug reports of Opensearch, Juju, Multipass, etc. (https://bugs.launchpad.net/juju/+bug/2074195, https://github.com/canonical/multipass/issues/3603, https://github.com/canonical/opensearch-operator/issues?q=is%3Aissue+author%3Ajuditnovak - more than 2 of those during this specific PR). Sometimes it feels that my own tasks are hardly progressing while providing (detailed, quality) bug reports for other tools :-(

Reply:
Agreed live: now we remove the check and in case we come across it again we'll investigate (open a bug, add a workaround, etc.)